read.go

```diff
@@ -231,7 +234,23 @@ func ReadFileInto(config interface{}, filename string) error {
 	if err != nil {
 		return err
 	}
+
+	// Skips a single leading UTF8 BOM sequence if it exists.
+	src = skipLeadingUtf8Bom(src)
+
 	fset := token.NewFileSet()
 	file := fset.AddFile(filename, fset.Base(), len(src))
 	return readInto(config, fset, file, src)
 }
+
+func skipLeadingUtf8Bom(src []byte) []byte{
+	if len(src) >= 3 {
```

Comment:
1. create global `var utf8Bom = []byte("\ufeff")`
2. use `len(utf8Bom)` instead of `3`
3. get rid of variable `bom`, join the two `if`s into one, and use `bytes.Equal` with utf8Bom for comparison